### DISCUSSION TITLE
write an timeout

### DISCUSSION THREAD

**Discussion by heislpeda** [jchristn/SuperSimpleTcp#87] [General]
hey, how can i write an timeout if the server doesnt send any data after 10 seconds, i tried everything but nothing worked :/

` private void timer1_Tick(object sender, EventArgs e)
        {
            if(eldisCLIENT.IsConnected == false)
            {
                eldisCLIENT.ConnectWithRetries(10);
                MessageBox.Show("no data.");
            }
        }`

**Comment by jchristn** [comment]
Hi @heislpeda - moving this to discussions.

**Reply by heislpeda** [reply]
Thanks. Can you tell me, how i can use your timeout function?

**Reply by jchristn** [reply]
Hi @heislpeda the timeout function is on the server side and not the client side.  i.e. the server disconnects a client if they haven't received any data in a given period of time.  There isn't a similar function native to the library on the client side at this time.  You could certainly build something outside of the library that mimics this behavior.  The ```ConnectWithRetries``` method is only for monitoring whether or not the client is able to successfully connect to the specified server within a given timeframe.

**Reply by jchristn** [reply]
Just integrated into the library, let me know how it works for you.

NuGet: https://www.nuget.org/packages/SuperSimpleTcp/2.4.1
Commit: https://github.com/jchristn/SimpleTcp/commit/f90bca5879a7bd38f9c4c29a41126610db6e911d
